common/src/main/java/org/apache/uniffle/common/compression/CompressionFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.compression;
+
+import org.apache.uniffle.common.config.RssConf;
+
+import static org.apache.uniffle.common.config.RssClientConf.COMPRESSION_TYPE;
+import static org.apache.uniffle.common.config.RssClientConf.ZSTD_COMPRESSION_LEVEL;
+
+public class CompressionFactory {
+
+  public enum Type {
+    LZ4,
+    ZSTD,
+    NOOP,
+  }
+
+  private CompressionFactory() {
+    // ignore
+  }
+
+  private static class LazyHolder {
+    static final CompressionFactory INSTANCE = new CompressionFactory();
+  }
+
+  public static CompressionFactory of() {
```

Comment:
Why is this method name is `of`?

Reply:
Emm. I refer to the implementation of flink, like this https://github.com/apache/flink/blob/7336672fa1e7b322122ff7ddc1e01227b716c070/flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogTable.java#L51

Reply:
Could we make it consistent with other factory methods in our project?

Reply:
Got it.

Reply:
Done. Rename to `getInstance`